Source/Parser/Functions/IterableJoiningFunction.cs

```diff
@@ -59,7 +59,10 @@ public override bool Evaluate(InterpreterScope scope, out ExpressionBase result)
                 iteratorScope.AssignVariable(predicateParameter, result);
 
                 if (!predicate.Evaluate(iteratorScope, out result))
+                {
+                    result = new ErrorExpression(string.Format("predicate function did not return any value for the predicate input {0}", input.ToString()));
```

Comment:
When creating an `ErrorExpression`, use the overload that takes an `ExpressionBase` whenever possible, as that will be used to define the token to underline for the error. In this case, `predicate` would make the most sense, but as suggested in other comments, the error should be set in `FunctionDefinition.Evaluate` and just be bubbled up here.